### ISSUE TITLE
[FEEDBACK] [Core] Support previous value

### ISSUE THREAD

**Issue by AlmogBaku** [raptor-ml/raptor#149]
### Background

I have a feature named `location`, and I want to implement a new feature that calculates the distance between the current location and the previous one.

To implement it, I need to some hacks which are not intuative.

### What do you propose to do?

Allow keeping `n` version in the State.

PyExp example:

```python
@raptor.register(float, staleness='24h', freshhnes='10h')
@raptor.keep_prev(1)
def lat(**req):
  return math.radians(req['payload']['lat'])

@raptor.register(float, staleness='24h', freshhnes='10h')
@raptor.keep_prev(1)
def lon(**req):
  return math.radians(req['payload']['lon'])

@raptor.Register('headless', 0,0)
def distance(**req):
  """distance in km"""
  
  lat1,_ = get_feature("lat")
  lon1,_ = get_feature("lon")
  lat2,_ =get_feature("lat[-1]")
  lon2,_ = get_feature("lon[-1]")

  dlon = lon2 - lon1
  dlat = lat2 - lat1
  
  # approximate radius of the earth in km
  R = 6371.0

  a = math.sin(dlat / 2)**2
  b = math.cos(lat1) * math.cos(lat2) * math.sin(dlon / 2)**2
  c = 2 * math.asin(math.sqrt(a+b))

  return = R * c
```

**Comment by AlmogBaku** [comment]
another way to solve the feature fetching, other than adding another "selector" to the FQN and making it an expression language,  we can call a unique function such as:

```python
value, timestamp = get_feature("lat", entity_id='almog', prev=-2)
```

**Comment by AlmogBaku** [comment]
closed by #241
